### PR TITLE
(#1399) refactor: voting function

### PR DESCRIPTION
**Pull request by rz-p**

### CODE REVIEW

**Comment by rz-p** [comment]
Vote count still not working

File: hms-app/src/components/card-details/VotingHandler.tsx [hunk 1]
```diff
@@ -84,7 +78,8 @@ export default function VotingHandler(props: IProps) {
             autoClose: 2000,
           })
         } else {
-          check(true)
+          check(false)
```
Comment (rz-p) [inline hms-app/src/components/card-details/VotingHandler.tsx:81]: changing from `true` to `false` makes the button behave correctly (which took me a long time to realize 😌)

File: hms-app/src/components/card-details/IdeaDetails.tsx [hunk 2]
```diff
@@ -207,7 +208,14 @@ export default function IdeaDetails(props: IProps) {
...
+      type === IdeaCardType.AllIdeas && <VoteButtons idea={props.idea} />
+    )
+  }
+
+  const voterCount = () => {
+    return (
+      { hackathonVotingOpened } &&
+      type === IdeaCardType.AllIdeas && <VoteList idea={props.idea} />
     )
   }
```
Comment (rz-p) [inline hms-app/src/components/card-details/IdeaDetails.tsx:220]: seperated vote count component for idea card

**Review by LetsPoke — APPROVED** [review]

File: hms-app/src/components/card-details/VoteList.tsx [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+  return (
+    <Card.Section className={classes.noBorderSection}>
+      <Stack align={'center'} spacing={'xs'}>
+        <Text className={classes.label}>Votes: </Text>
+        <Text className={classes.text}>{idea.voters?.length}</Text>
+      </Stack>
+    </Card.Section>
+  )
+}
```
[the remark this answers is not in the document]
Comment (LetsPoke) [reply hms-app/src/components/card-details/VoteList.tsx:21]: yeah its a bit short

**Pull request merged**